main.go

```diff
@@ -29,6 +30,11 @@ func main() {
 		panic(err)
 	}
 
+	if err := db.InitializeDB(); err != nil {
+		panic(err)
+	}
+	defer db.CloseDB()
```

Comment:
This should probably be run in a different goroutine here. Also it might be good to add an option to enable/disable the pinning system in the config